app/views/posts/index.html.erb

```diff
@@ -1,8 +1,57 @@
-<h2>User name</h2>
-<span>Number of posts:</span>
-<h3>Post #1</h3>
-<p>Some text</p>
-<h4>User comment: 1</h4>
-<h4>User comment: 2</h4>
-<h4>User comment: 3</h4>
-<h4>User comment: 4</h4>
\ No newline at end of file
+<div class="d-flex flex-column align-items-center gap-4 my-5">
+    <div class="container d-flex w-75 p-0">
+        <a href="<%= user_path(@user) %>"><img class="rounded-circle bg-white p-2 border me-4" src="<%= @user.photo %>" alt="User Photo"></a>
+        <div class="border rounded-5 w-100 bg-white p-4 d-flex flex-column justify-content-between">
+            <div class="w-25 m-0">
+                <h2 class="text-decoration-none fs-2 text-decoration-underline-hover text-primary"><%= link_to @user.name, user_path(@user), class: "text-decoration-none" %></h2>
+                <p class="user-posts text-end me-5 fs-5">Number of posts: <%= @user.posts_counter %></p>
+            </div>
+            <%= link_to "Back to Home", root_path, title:"To Home Page", class: "text-decoration-none fs-4 text-decoration-underline-hover text-primary align-self-end" %>
+        </div>
+    </div>
+
+    <div class="d-flex flex-column align-items-center gap-4 w-75">
+        <div class="d-flex flex-column align-items-center gap-4 w-100">
+        
+            <% @posts.each do |post| %> 
+                <div class="border rounded-5 w-100 bg-white p-4">
+                    <h3 class="mb-0">
+                            <% if post.comments.count > 0 %>
+                                <%= link_to post.title + @user.name,  user_post_path(user_id: @user.id, id: post.id), class: "text-decoration-none text-decoration-underline-hover" %>
+                            <% else%>
+                                <%= post.title + @user.name %>
+                            <% end %>
+                        </h3>
+                    <span class="text-secondary">
+                        Post no. <%= post.id %>
+                    </span>
+                    <p class="fs-3 my-3"><%= post.text %></p>
+                    <div class="text-end">
+                        <p class="mb-0 fs-5"><%= post.comments.count %> Comments | <%= post.likes.count %> Likes</p>
+                    </div>
+                    <% if post.comments.count > 0 %>
+                        <div class="border w-75 mx-auto m-4 p-0 align-self-center"></div>
+                        <h3 class="text-center m-0">Recent Comments</h3>
+                        <div class="d-flex flex-column gap-3">
+                        <% post.comments.limit(5).each do |comment| %>
+                            <div class="d-flex flex-column">
+                                <h4 class="mb-0">
```

Comment:
- [ ] Though the app works as expected, the recent_comments function in the post model should have been used to return 5 recent comments instead of creating a different loop (code). Kindly make use of this function because this is the main reason it was created.